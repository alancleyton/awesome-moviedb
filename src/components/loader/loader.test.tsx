import { render, screen } from '@testing-library/react';

import { Loader } from '.';

describe('Loader', () => {
  it('should render correctly with children', () => {
    const { container } = render(
      <Loader>
        <h2>loader</h2>
      </Loader>,
    );

    expect(screen.getByText(/loader/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render loader when isLoader=true', () => {
    render(
      <Loader isLoading>
        <h2>loader</h2>
      </Loader>,
    );

    expect(screen.queryByText(/loader/i)).not.toBeInTheDocument();
    expect(screen.getByRole('alert')).toBeInTheDocument();
  });
});
