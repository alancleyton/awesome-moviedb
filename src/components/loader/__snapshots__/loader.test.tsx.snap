// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly with children 1`] = `
<div>
  <h2>
    loader
  </h2>
</div>
`;
